### PR TITLE
[REF] ScheduleReminders - Cleanup "sublimely stupid" form code

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Initial cleanup of the ScheduleReminders form... toward sanity.
Lots more refactoring to do, this fixes the easy-to-fix things that were obviously wrong.

### CODE REVIEW

File: CRM/Admin/Form/ScheduleReminders.php [hunk 1]
```diff
@@ -138,17 +146,12 @@ public function buildQuickForm(): void {
       }
     }
     else {
-      // Dig deeper - this code is sublimely stupid.
```
Comment (colemanw) [inline CRM/Admin/Form/ScheduleReminders.php:141]: This is easily my favorite comment in our codebase. I'm sorry to see it go @totten

**Comment by seamuslee001** [comment]
Jenkins re test this please

**Comment by colemanw** [comment]
@seamuslee001 was that just a drive-by comment or were you gonna merge this?

**Comment by seamuslee001** [comment]
ok this seems ok

File: CRM/Admin/Form/ScheduleReminders.php (hunk 1 above)
Comment (totten) [reply CRM/Admin/Form/ScheduleReminders.php:141]: Well, I guess that code had to wander 7 years through the wilderness, but now... it's sublimely smart... I hope...

**Comment by totten** [comment]
I played around with it a bit on the test site -- it exhibited consistency in writing/reading the "Entity"-related options. So that seems good.

My process was to make one "Scheduled Reminder", then repeatedly edit it (changing the "Entity" to some random values). There were a few PHP warnings. I haven't tested carefully enough to see if the originate here.

Screen Shot 2023-07-19 at 4 01 46 PM

Screen Shot 2023-07-19 at 4 01 54 PM

**Comment by colemanw** [comment]
@totten yea, that's because the form has to do its own deserialization of the values from `BAO::retrieve` because that function sucks. I'll get to that...

**Pull request merged**